pkg/operator/resource/resourceapply/core.go

```diff
@@ -234,10 +234,18 @@ func ApplyConfigMap(client coreclientv1.ConfigMapsGetter, recorder events.Record
 }
 
 // ApplySecret merges objectmeta, requires data
-func ApplySecret(client coreclientv1.SecretsGetter, recorder events.Recorder, required *corev1.Secret) (*corev1.Secret, bool, error) {
-	if len(required.StringData) > 0 {
-		return nil, false, fmt.Errorf("Secret.stringData is not supported")
+func ApplySecret(client coreclientv1.SecretsGetter, recorder events.Recorder, requiredInput *corev1.Secret) (*corev1.Secret, bool, error) {
+	// copy the stringData to data.  Error on a data content conflict inside required.  This is usually a bug.
+	required := requiredInput.DeepCopy()
+	for k, v := range required.StringData {
+		if dataV, ok := required.Data[k]; ok {
+			if string(dataV) != v {
+				return nil, false, fmt.Errorf("Secret.stringData[%q] conflicts with Secret.data[%q]", k, k)
```

Comment:
the api docs say this always overrides data. if you were to change a field in the code using stringData in the code (I don't see why), wouldn't this break the update?

Reply:
> the api docs say this always overrides data. if you were to change a field in the code using stringData in the code (I don't see why), wouldn't this break the update?

I think that would mean inconsistent information was provided as requiredInput, so a failure here is reasonable.